### PR TITLE
`theme pull/push`: Pass development theme from CLI 3’s to CLI 2’s local storage

### LINKED ISSUE #1387: `theme pull` (CLI 2) does not show development theme (created by CLI 3) in the select list
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme, Other

### Expected behavior

Normally we work on the theme editor of our development theme and pull the changes to our local repo. but. Theme pull should show the development theme in the list as it was done in earlier versions of Shopify cli 3.x.the 

### Actual behavior

I've initialized them using theme init in my vs code. Now I run the Shopify theme dev command and open the link to Customize this theme in the Theme Editor. I want to pull the changes I've made in my theme editor. but when I run the Shopify theme pull command it shows me a list of themes in my store and I have to choose one of them to pull. but I don't want that I want to pull changes I've made in my development theme but it is not on the list.

### Stack trace

_No response_

### Reproduction steps

1. run Shopify theme init 
2. run Shopify theme dev and open theme customizer and do some changes in the layout.
3. now run Shopify pull command we can't pull these changes as the theme is not in the list given to pull.

### Operating System

Windows 11

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.43

### Shell

powershell

### Node version (run `node -v` if you're not sure)

18.12.1

### What language and version are you using in your application?

_No response_

### PR DESCRIPTION
**Pull request by Poitrin**
### WHY are these changes introduced?

Fixes #1387

### WHAT is this pull request doing?

* Retrieves, when `shopify theme pull` or `… push` is called, development theme ID, if existing, and passes it from CLI 3 to 2, so that CLI 2 can set the development theme ID in its own local storage (`ShopifyCLI::DB`). That way, development theme is **not** treated as [_foreign development_](https://github.com/Shopify/cli/blob/c8ece7a60b9ac88bb56329b46241f866787dbd99/packages/cli-kit/assets/cli-ruby/lib/project_types/theme/forms/select.rb#L17) and therefore **not** hidden from the select list anymore.
* Displays `[yours]` label in `shopify theme open` list, if applicable.

### How to test your changes?

Note: `(pnpm )shopify:run` → CLI 3, `shopify-dev` → CLI 2

#### `theme pull`

1. `pnpm build`
2. Verify that `shopify:run theme list`’s and `shopify-dev theme list`’s `[yours]` development themes have different IDs, if development themes exist.
3. `shopify:run theme init`, with name e.g. `draft-1`
4. `shopify:run theme dev --path="./draft-1"`
5. Note down the ID, and wait until command is finished
6. `shopify:run theme pull --path="./draft-2"`
7. Verify that the `pull` command lists a development theme with `[yours]` label. Note down the name and Ctrl+C.
8. Verify that `shopify:run theme list` and `shopify-dev theme list` both list the `[yours]` development theme, with the ID and name you noted down before.

#### `theme push`
1. `shopify-dev theme delete <Theme ID of the development theme you just created>`
2. Verify that `shopify-dev theme list` does not list the theme anymore
3. `shopify:run theme dev --path="./draft-1"` and note down the ID of the new development theme
4. `shopify:run theme list` → Note down the name of the development theme
5. `shopify:run theme push --path="./draft-1"` and verify that the development theme with the name you noted down is listed

#### `theme open`
1. `shopify:run theme open`
2. Verify that the development theme you created is tagged with `[development] [yours]`.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/theme/src/cli/commands/theme/pull.ts [hunk 1]
```diff
@@ -54,20 +54,21 @@ export default class Pull extends ThemeCommand {
...
     const adminSession = await ensureAuthenticatedThemes(store, flags.password)
 
-    if (flags.development) {
-      const theme = await new DevelopmentThemeManager(adminSession).find()
-      flags = {
-        ...flags,
-        development: false,
-        theme: theme.id.toString(),
+    const developmentThemeManager = new DevelopmentThemeManager(adminSession)
+    const theme = await (flags.development ? developmentThemeManager.find() : developmentThemeManager.fetch())
```
Comment (mgmanzella) [inline packages/theme/src/cli/commands/theme/pull.ts:65]: maybe im missing something but does this mean even when a developer uses theme pull to get a live theme, the `developmentThemeManager` is used?

File: packages/theme/src/cli/services/open.ts [hunk 2]
```diff
@@ -10,17 +10,16 @@ export async function open(
...
 ) {
-  let themeId = options.theme
-  if (options.development) {
-    const theme = await new DevelopmentThemeManager(adminSession).find()
-    themeId = theme.id.toString()
-  }
-
+  const developmentThemeManager = new DevelopmentThemeManager(adminSession)
+  const developmentTheme = (
+    await (options.development ? developmentThemeManager.find() : developmentThemeManager.fetch())
```
Comment (mgmanzella) [inline packages/theme/src/cli/services/open.ts:15]: this branching logic to decide which method to use on the `developmentThemeManager` is appearing a lot, which feels a bit like a leaky abstraction 🤔 do you think it's possible and sensible to move this logic into the development theme manager?

**Review by mgmanzella — COMMENTED** [review]
noice!! had a couple questions

File: packages/theme/src/cli/commands/theme/pull.ts (hunk 1 above)
Comment (Poitrin) [reply packages/theme/src/cli/commands/theme/pull.ts:65]: Great question!
Yes, we pass the `--development-theme-id` every time a Development Theme exists, to make sure that CLI 2 always has the most recent ID in its DB (see description).
The `--theme` flag can still include the live theme though :)

File: packages/theme/src/cli/services/open.ts (hunk 2 above)
Comment (mgmanzella) [reply packages/theme/src/cli/services/open.ts:15]: yehhhh i see what you're saying, the thing that feels different in this scenario is the call site is deciding which method to use on DevelopmentManager based on an option/flag that could just be passed to the constructor. and every case ive seen the default is to use fetch() if development id isn't set

im cool to leave as is tho 👍

**Review by mgmanzella — APPROVED** [review]
nice job!! tophatting looked great, thank you @Poitrin!

**Pull request merged**